### ISSUE TITLE
Reference Error: Can't find variable: Set

### ISSUE THREAD

**Issue by mdominado** [ariya/phantomjs#15002]
1. Which version of PhantomJS are you using? Tip: run `phantomjs --version`.
 - 2.1.1
2. What steps will reproduce the problem?
 - Used the github.js sample code from the main site

Example:
```
var page = require("webpage").create();
page.open("https://developers.arcgis.com/javascript/latest/sample-code/widgets-print/live/index.html", function(status) {
	if(status == "success"){
		page.render("web.png");
	} else {
		console.log("cant load page");
	}
	
	phantom.exit();

});
```
 - Replaced the url parameter that has the github url to: https://developers.arcgis.com/javascript/latest/sample-code/widgets-print/live/index.html
- Ran it locally: phantomjs (wherever github.js is on my machine)

3. Which operating system are you using?
- Windows 10
4. Did you use binary PhantomJS or did you compile it from source?
- Binary PhantomJS
5. Please provide any additional information below.
It returns: ReferenceError: Can't find variable: Set. The above sample is not my web page, but it is using a Javascript API provided by ESRI, which my web page is also using. So, it is throwing the same error.

**Comment by kensoh** [comment]
@mdominado I'm a CasperJS user for couple of years, now browsing through open issues on CasperJS/PhantomJS. The JS looks minified to try to decode, looks like issue happens elsewhere so wanted to link this - https://github.com/giakki/uncss/issues/304

Sounds like PhantomJS version not supporting Set? - https://stackoverflow.com/questions/38051320/unable-to-find-element-using-selenium-and-phantomjs

**Comment by mdominado** [comment]
@kensoh 

Yeah the code is definitely minified, so all I can tell is that it wants to use a Set to do something. In any case, I agree with you on the Set not being supported by PhantomJS, that was one of my guesses, but I hoped that wasn't going to be the case.

I assume there is no way around it at this point then. Until PhantomJS supports the newer EMCAScript standards, there is not really much I can do. Especially, considering that there might be more functionalities from the newer ES6 standard that the API I'm using has implemented.

**Comment by kensoh** [comment]
@mdominado actually does anyone know the state of things for PhantomJS? After maintainer Vitaly's announcement in April that he is stepping down. And Chrome headless is coming, Firefox headless is coming.

I really hope and still believe someone or organization would step up to take on the baton. After all, so much vested interests have been made in the PhantomJS ecosystem. Or at least some form of mitigation steps to ease any transition.
